Do not store build host name

bmwiedemann opened the pull request:
* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

Do not store build host name
if reproducible builds are wanted.
See https://reproducible-builds.org/ for why this is good.

This affected scons itself, which differed in the line

```python
__buildsys__ = "..."
```

bdbaddog commented:
Looks good. Merging.

bmwiedemann commented:
So far, everyone wanting reproducible build results sets SOURCE_DATE_EPOCH - defined in https://reproducible-builds.org/specs/source-date-epoch/ ; adding another variable would be possible but mean extra effort for people to discover and set it.

mwichmann commented:
If I'm getting this, this patch proposes, for the construction of an scons release, to honor SOURCE_DATE_EPOCH if it is set, as a flag for not deriving the build host, but not otherwise used.  Shouldn't it also be used to fill in the date field, which is currently the time scons was run, and thus will vary?  Not sure I think "reproducible" is the best magic string, maybe with a leading underscore to mark it as special?   It also looks like the same can be achieved by externally setting BUILD_SYSTEM to the magic string, though as noted, that's something extra for people to discover.  As one more nit, at the very least some more docu is needed on this... for example, if you don't pass in DEVELOPER it will still be set by inspection, and thus give you different results if different people do the build.

bmwiedemann commented:
DATE is already using SOURCE_DATE_EPOCH since #3221 

I updated the commit to use _reproducible to mark it as more special and added a 2nd commit for DEVELOPER

bdbaddog commented:
> So far, everyone wanting reproducible build results sets SOURCE_DATE_EPOCH - defined in https://reproducible-builds.org/specs/source-date-epoch/ ; adding another variable would be possible but mean extra effort for people to discover and set it.

O.k. If that's the common practice. I'll go ahead and merge.

mwichmann commented:
Ah, I missed the earlier change.  Looking okay to me, but we'll wait for the Boss :)

Does scons need to be taught anything as far as producing reproducible builds? (as opposed to the build of scons itself)  Unrelated, just curious.
